hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java

```diff
@@ -76,10 +76,10 @@ public class ResourceHandlerModule {
   private static volatile CpuResourceHandler
       cGroupsCpuResourceHandler;
 
-  private static void initializeCGroupHandlers(Configuration conf)
-      throws ResourceHandlerException {
+  private static void initializeCGroupHandlers(Configuration conf,
+      CGroupsHandler.CGroupController controller) throws ResourceHandlerException {
     initializeCGroupV1Handler(conf);
-    if (cgroupsV2Enabled) {
+    if (cgroupsV2Enabled && !isMountedInCGroupsV1(controller)) {
       initializeCGroupV2Handler(conf);
```

Comment:
Just a thought: do we need to initialize the v1 handler if we're looking at clean v2 case?

Reply:
Thanks for the review @brumi1024! With the current logic we use the v1 handler to get the v1 controller paths and init v2 handler only if the given controller is not mounted in v1. But with a pure v2 setup I don't think we will use v1 handler after that check, there may be a cleaner way to determine this.

It's not the clean v2 setup, but with the resource plugins we still use only the v1 handler in [this](https://github.com/apache/hadoop/pull/7222) codepart, so probably additional checks would be needed there as well if we decide not to init the v1 handler.